packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -388,3 +397,66 @@ export function mapOpenSeaContractV2ToV1(
     },
   };
 }
+
+/**
+ * Retrieves token prices for a set of contract addresses in a specific currency and chainId.
+ *
+ * @param args - The arguments to function.
+ * @param args.tokenPricesService - An object in charge of retrieving token prices.
+ * @param args.nativeCurrency - The native currency to request price in.
+ * @param args.tokenAddresses - The list of contract addresses.
+ * @param args.chainId - The chainId of the tokens.
+ * @returns The prices for the requested tokens.
+ */
+export async function fetchTokenContractExchangeRates({
```

Comment:
Have you considered incorporating this into the token price service itself? That would let us remove the duplication between here and the controller.

Reply:
Sorry — this was my recommendation to bring over the batching code from the controller. My thought was that we could refactor the controller later to use this function, but at least at the moment, if the batching were in this function too, the behavior would be the same regardless of whether you used this function or the controller. Wondering if we could take care of moving the batching in another PR? That should be a non-breaking change.

Reply:
Well, it might result in breaking changes, depending on how we merge the two. If we move it into the service, the other methods on the serve might no longer be needed, letting us remove those from the API. Though of course we could do that in two steps.

And overall yes I agree, we can merge this and refactor it later.

Reply:
Just created an issue to track this suggestion: https://github.com/MetaMask/core/issues/4117